### DISCUSSION TITLE
Add CORS headers to Ray App

### DISCUSSION THREAD

**Discussion by patinthehat** [spatie/ray#157] [Ideas]
I'm working on something that sends data to Ray from the browser, however due to a missing `Access-Control-Allow-Origin` header, I cannot send anything due to the CORS restrictions.

Can you add an option in preferences to modify this, or add `Access-Control-Allow-Origin: *` to the headers in the app's Express server?

Error message:
```console
Access to XMLHttpRequest at 'http://127.0.0.1:23517/' from origin 'http://localhost:3344' 
has been blocked by CORS policy:

Response to preflight request doesn't pass access control check: 
No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by patinthehat** [comment]
The workaround I've come up with in the mean time is to run an http proxy that adds the required CORS headers.
You can check it out at [permafrost-dev/ray-proxy](https://github.com/permafrost-dev/ray-proxy).

**Comment by AdrianMrn** [comment]
What version of the app are you on? I added the required headers using the `cors` middleware in 1.2.5 or so.

**Comment by patinthehat** [comment]
Ray version: 1.3.0
OS: Ubuntu 20.04
Browser: Chromium 87

When I dump the headers from a `POST` request to Ray using curl, here's what I get:
```http
HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: text/html; charset=utf-8
Content-Length: 2
ETag: W/"2-eoX0dku9ba8cNUXvu/DyeabcC+s"
Date: Thu, 14 Jan 2021 13:31:11 GMT
Connection: keep-alive

ok
```

and the headers from an `OPTIONS` request:

```http
HTTP/1.1 200 OK
X-Powered-By: Express
Allow: POST
Content-Type: text/html; charset=utf-8
Content-Length: 4
ETag: W/"4-Yf+Bwwqjx254r+pisuO9HfpJ6FQ"
Date: Thu, 14 Jan 2021 13:34:09 GMT
Connection: keep-alive

POST
```

**Comment by AdrianMrn** [comment]
That's weird 😅 

![image](https://user-images.githubusercontent.com/12762044/104605333-a3dd7600-567e-11eb-8957-f92355ecbdbe.png)
![image](https://user-images.githubusercontent.com/12762044/104605356-afc93800-567e-11eb-9055-c5aaf16d082e.png)

**Comment by patinthehat** [comment]
Strange.

Is it possible you're running a development version? (Or on a different OS?)
Could you list the express/cors package versions you're using?

**Comment by AdrianMrn** [comment]
express@4.17.9
cors@2.8.5

I'm on Mac running a production version of Ray 1.3.0. I'm not sure what's going wrong 😬

I'm trying to find a solution.

**Comment by AdrianMrn** [comment]
Oh boy.

![image](https://user-images.githubusercontent.com/12762044/104616942-2e2bd700-568b-11eb-8a52-5f7b2d31121a.png)

It'll be included in the next version 🙄

**Comment by patinthehat** [comment]
Fantastic. Thanks for looking into it!
